application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/BeanConfigurator.java

```diff
@@ -106,6 +138,20 @@ private static Config getConfiguration() throws ConfigurationException
         }
     }
 
+    private static <T> T getConfiguration(InputStream configurationFile, Class<T> clazz) throws ConfigurationException
+    {
+        try
+        {
+            ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+
+            return objectMapper.readValue(configurationFile, clazz);
+        }
+        catch (IOException e)
+        {
+            throw new ConfigurationException("Unable to load configuration file " + configurationFile, e);
```

Comment:
This IO exception will become quite odd now.. (shouldn't happen though, so it shouldn't cause any issues)